### DISCUSSION TITLE
Coder-app share level not respected when using wildcard access URL

### DISCUSSION THREAD

**Discussion by ken-schu** [coder/coder#13889] [General] [answered]
Hey all -

Thanks for all the thoughtfulness around port forwarding/sharing. 

We have a use case where we share code-server publicly via #4378 and it works well - we were able to view forwarded ports via path publicly.

But we wanted to serve apps on forwarded ports, so following the advice of [this guide](https://coder.com/docs/code-server/guide#proxying-to-create-a-react-app), we set `CODER_WILDCARD_ACCESS_URL` and now use subdomains as the default generated port share link. These links now require authentication though - and we can't find a way to share them publicly.

Is this a known limitation? Or am I missing some configuration to share this kind of port forward publicly as well? I was hoping that the coder app share level would be respected in #4378 - but it doesn't seem like that's the case.

Thanks!

**Comment by ken-schu** [comment]
I believe [this](https://github.com/coder/code-server/discussions/6151#discussioncomment-5712031) is what I'm trying to do and its not possible. Although I thought that if I had `share = public` on my code-server, that'd be respected on port forwarding too.

cc @code-asher as you seem to be thinking about this right now

**Reply by code-asher** [reply]
Are you running into the code-server authentication page?  When using code-server with Coder you should launch code-server with `--auth none`; Coder already does auth so there is no need to use code-server's.

**Reply by ken-schu** [reply]
No, running into the Coder OSS auth page:

<img width="644" alt="Screenshot 2024-07-15 at 5 44 29 PM" src="https://github.com/user-attachments/assets/79b7c12d-e03a-4a7b-918d-63aa5faa8273">

I launch code-server with no auth:
` /tmp/code-server/bin/code-server --disable-workspace-trust --auth none --port 13337 >/tmp/code-server.log 2>&1 &`

And code_server resource is public:

```
resource "coder_app" "code-server" {
  agent_id     = coder_agent.main.id
  slug         = "code-server"
  display_name = "code-server"
  url          = "http://localhost:13337/?folder=/home/${local.username}"
  icon         = "/icon/code.svg"
  subdomain    = true
  share        = "public"

  healthcheck {
    url       = "http://localhost:13337/healthz"
    interval  = 5
    threshold = 6
  }
}
```

Wildcard access URL is also set. Is that expected behavior?

**Reply by code-asher** [reply]
Ahhhh OK, I misunderstood a bit.  In this case code-server's proxy is not involved at all.  Coder's wildcard proxy forwards directly to the associated port on the workspace, not to a specific app, so the code-server application running on the workspace would never see the request at all.

But there is no need to use code-server's proxy anyway because Coder's does let you share individual ports.  In the dashboard in each workspace there is a ports dropdown where you can select which ports to share.  See https://coder.com/docs/networking/port-forwarding#sharing-ports

That means you could also remove the share on the code-server app as well, unless you need code-server also to be public.

**Reply by code-asher** [reply, marked as answer]
Well, my comment there is about code-server with auth on, but you have auth off and are not hitting code-server's sub-domain proxy anyway, so my comment would not apply here (although it does sound very similar to the Coder case).

Keep in mind that Coder and code-server are two separate unrelated applications that both just happen to have their own proxies.  There is no way to make Coder's workspace port proxy forward all requests to that workspace to some other proxy/app running in that workspace (code-server in this case).

But yeah, Coder does automatically forward ports but the share level always defaults to `owner` as far as I know.  It could maybe make sense to submit a feature request for changing the default share level of a port.  It does sound a little risky though.  :stuck_out_tongue:

I hope that makes sense!
